pkg/startup/v7/derive.go

```diff
@@ -135,3 +139,216 @@ func (derivedType) CaBundle(cs *api.OpenShiftManagedCluster) ([]*x509.Certificat
 
 	return tls.UniqueCert(caBundle), nil
 }
+
+// TDAgentDotConf returns /etc/td-agent/td-agent.conf content
+func (derivedType) TDAgentDotConf() (string, error) {
+	// so far, this is a static config, may change later
+	return `<source>
+  @type systemd
+  <storage>
+    @type local
+    path /var/run/td-agent/journald-cursor.json
+  </storage>
+  tag journald
+</source>
+<source>
+  @type tail
+  format json
+  path /var/log/openshift-audit/*.log
+  pos_file /var/run/td-agent/openshift-audit.pos
+  refresh_interval 10
+  time_key timestamp
+  time_format %Y-%m-%dT%H:%M:%SZ
+  tag k8s_audit
+</source>
+<filter journald>
+  @type record_transformer
+  enable_ruby true
+  <record>
+    MESSAGE ${record["MESSAGE"].nil? ? nil : record["MESSAGE"].force_encoding("UTF-8").encode("ASCII", invalid: :replace, undef: :replace)}
+    # data format:
+    # k8s_apiserver_apiserver-vqxg4_kube-service-catalog_72ce4d73-5224-11e9-98d0-000d3a196756_0
+    CONTAINER ${record["CONTAINER_NAME"].nil? ? nil : record["CONTAINER_NAME"].split("_")[1] }
+    POD ${record["CONTAINER_NAME"].nil? ? nil : record["CONTAINER_NAME"].split("_")[2] }
+    NAMESPACE ${record["CONTAINER_NAME"].nil? ? nil : record["CONTAINER_NAME"].split("_")[3] }
+    CONTAINER_ID ${record["CONTAINER_NAME"].nil? ? nil : record["CONTAINER_NAME"].split("_")[4] }
+  </record>
+</filter>
+<match **>
+  @type forward
+  send_timeout 60s
+  recover_wait 10s
+  hard_timeout 60s
+
+  <server>
+    name localhost
+    host 127.0.0.1
+    port 5001
+    weight 60
+  </server>
+</match>`, nil
+}
+
+// MDSDDotXML returns /etc/mdsd.d/mdsd.xml config contents
+func (derivedType) MDSDDotXML(cs *api.OpenShiftManagedCluster) (string, error) {
+	var tmpl = `<?xml version="1.0" encoding="utf-8"?>
```

Comment:
I am not a fan of this massive multi-line static config in the middle of a function... what's the idiomatic Golang way to handle this? I'd be inclined to read it from a file or at least put it at the top of this file.

Reply:
discussed on Slack, a better direction would be to get the text under data/ ; use a common/ directory to limit duplication across master/ and worker/